aws/resource_aws_iam_role.go

```diff
@@ -239,6 +264,21 @@ func resourceAwsIamRoleUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("max_session_duration") {
+		roleMaxDurationInput := &iam.UpdateRoleInput{
+			RoleName:           aws.String(d.Id()),
+			MaxSessionDuration: aws.Int64(d.Get("assume_role_policy").(int64)),
```

Comment:
This looks like it has two issues:
* Copy-paste typo of which attribute to grab, should be `"max_session_duration"`
* `schema.TypeInt` returns a regular `int`, so I believe this needs to be `aws.Int64(int64(d.Get("max_session_duration").(int)))`